Create hc_waste_card.yaml

ptC7H12 opened the pull request.

damianeickhoff commented:
Its a nice touch to the overall theme and a good alternative for the bigger card.

Though, the card is not dynamic enough. Atleast, my entites won't work. I have to be honest, I don't really understand this part
```yaml
    z: |
      [[[
        const z = variables.trash_entities.map(entity => ({
          entity: entity.entity,
          name: entity.name,
          icon: entity.icon,
          color: entity.color,
          days_left: states[entity.entity] ? parseInt(states[entity.entity].state, 10) : 999
        }));
        // Sort for left days
        return JSON.stringify(z.sort((a, b) => a.days_left - b.days_left));
      ]]]
```
So I don't know what is wrong. Most logical reason is maybe the way your sensors handle dates compared to mine:

![image](https://github.com/user-attachments/assets/95f96ff3-d32e-4107-a377-b9db08fd6c7c)

Wondering if you can make it more dynamic so it also suites my kind of sensors.

ptC7H12 commented:
@damianeickhoff I've changed the code a little bit.

I've changed the Entity to a "days" variable...

```
  #########################################################################
  # EXAMPLE
  #- type: custom:button-card
  #  template: hc_waste_card
  #  variables:
  #    static_text: "Tage bis zur Abholung vom"
  #    trash_entities:
  #      - days: |
  #          [[[ return states['sensor.restmull'].state || 0 ]]]
  #        name: Restmüll
  #        icon: far:trash-can
  #        color: rgba(141, 123, 123, 1)
  #      - days: |
  #          [[[ return states['sensor.papierbehalter'].state || 0 ]]]
  #        name: Papiermüll
  #        icon: far:newspaper
  #        color: rgba(141, 174, 202, 1)
  #      - days: |
  #          [[[ return states['sensor.wertstofftonne'].state || 0 ]]]
  #        name: Wertstoff
  #        icon: fas:recycle
  #        color: rgba(151, 207, 201, 1)
  #########################################################################
```

In the days variable you have to pass your `Days_until` attribute (a numeric value) - for me it's the state.
